Jenkinsfile

```diff
@@ -79,6 +119,12 @@ pipeline {
 		}
 	}
 	post {
+		success {
+			recovery()
+		}
+		failure {
+			fail()
```

Comment:
Move the content of the `fail` function here for clarity (it's only called here)